src/sentry/rules/actions/sentry_apps/notify_event.py

```diff
@@ -146,3 +154,15 @@ def after(self, event: GroupEvent, state: EventState) -> Generator[CallbackFutur
             sentry_app=sentry_app,
             schema_defined_settings=self.get_option("settings"),
         )
+
+    def render_label(self) -> str | None:
+        sentry_app_installation_uuid = self._get_sentry_app_installation_uuid()
+
+        try:
+            sentry_app = SentryApp.objects.get(installations__uuid=sentry_app_installation_uuid)
+        except SentryApp.DoesNotExist:
+            raise ValidationError("Could not identify integration from the installation uuid.")
+
+        alert_rule_component = self._get_alert_rule_component(sentry_app.id, sentry_app.name)
+
+        return alert_rule_component.schema.get("title")
```

Comment:
Here is where `title` is returned to be used as the label and thus as the name in the UI